### ISSUE TITLE
please improve the form output's markup 

### ISSUE THREAD

**Issue by alsd** [jameslafferty/MailChimp-Widget#11]
WordPress core has pretty good accessibility but IMHO the WordPress plugins are quite the oppsitve, including this nice plugin!

I am working a site that needs to be WCAG 2.0 AA compliance, and the output of your codes are missing IDs, and label that needs to associates to each ID in the input attribute.

I clean up your code, and you please upgrade it so that I don't need to clean up again if there is an update. Thanks!

lib/ns_widget_mailchimp.class.php

starts from line #288 to #300 - what I did is insert the code used in the name  for label and add ID to it.

Each label must associate to an ID in the input: example: 
&lt;label for=&quot;id&quot;&gt;xxx&lt;label&gt;
&lt;input name=&quot;x&quot;  id=&quot;id&quot; &gt;

$collect_first = '&lt;div&gt;&lt;label for=&quot;' . $this-&gt;id_base . '_first_name&quot;&gt;' . __('First Name :', 'mailchimp-widget') . '&lt;input type=&quot;text&quot; name=&quot;' . $this-&gt;id_base . '_first_name&quot; id=&quot;' . $this-&gt;id_base . '_first_name&quot; /&gt;&lt;/label&gt;&lt;/div&gt;';

```
            }

            $collect_last = '';

            if ($instance['collect_last']) {

                $collect_last = '&lt;div&gt;&lt;label for=&quot;' . $this-&gt;id_base . '_last_name&quot;&gt;' . __('Last Name :', 'mailchimp-widget') . '&lt;input type=&quot;text&quot; name=&quot;' . $this-&gt;id_base . '_last_name&quot; id=&quot;' . $this-&gt;id_base . '_last_name&quot; /&gt;&lt;/label&gt;&lt;/div&gt;';

            }

            $widget .= '&lt;form action=&quot;' . $_SERVER['REQUEST_URI'] . '&quot; id=&quot;' . $this-&gt;id_base . '_form-' . $this-&gt;number . '&quot; method=&quot;post&quot;&gt;' . $this-&gt;subscribe_errors . $collect_first . $collect_last . '&lt;div&gt;&lt;label for=&quot;' . $this-&gt;id_base . '_email&quot;&gt;' . __('Email Address :', 'mailchimp-widget') . '&lt;/label&gt;&lt;input type=&quot;hidden&quot; name=&quot;ns_mc_number&quot; value=&quot;' . $this-&gt;number . '&quot; /&gt;&lt;input type=&quot;text&quot; name=&quot;' . $this-&gt;id_base . '_email&quot;  id=&quot;' . $this-&gt;id_base . '_email&quot;/&gt;&lt;/div&gt;&lt;div&gt;&lt;input class=&quot;button&quot; type=&quot;submit&quot; name=&quot;' . __($instance['signup_text'], 'mailchimp-widget') . '&quot; value=&quot;' . __($instance['signup_text'], 'mailchimp-widget') . '&quot; /&gt;&lt;/div&gt;&lt;/form&gt;&lt;script type=&quot;text/javascript&quot;&gt; jQuery(\'#' . $this-&gt;id_base . '_form-' . $this-&gt;number . '\').ns_mc_widget({&quot;url&quot; : &quot;' . $_SERVER['PHP_SELF'] . '&quot;, &quot;cookie_id&quot; : &quot;'. $this-&gt;id_base . '-' . $this-&gt;number . '&quot;, &quot;cookie_value&quot; : &quot;' . $this-&gt;hash_mailing_list_id() . '&quot;, &quot;loader_graphic&quot; : &quot;' . $this-&gt;default_loader_graphic . '&quot;}); &lt;/script&gt;';
```

Somone commented onthe poor HTML markup,  thought I add my 2 cents too. <br/> in the form element  isn't as terrible comparing with what I saw often that people use it for paragraph to gain line break/paddings.  In the above codes, I removed <br/> and replace it with <div> tag instead.

The reason is this give extra flexibility for web designers to style their desired layouts, for example,  I want first name, last name fields display in two columns floated, and I don't want the label and input field be aligned (with your codes this is still doable by declaring display block or clear both).

e.g 
&lt;div&gt;
&lt;label for=&quot;first-name&quot;&gt;First Name&lt;/label&gt;
&lt;input name=&quot;first-name&quot; id=&quot;first-name&quot; type=&quot;text&quot;/&gt;
&lt;/div&gt;

&lt;div&gt;
&lt;label for=&quot;last-name&quot;&gt;Last Name&lt;/label&gt;
&lt;input name=&quot;last-name&quot; id=&quot;last-name&quot; type=&quot;text&quot;/&gt;
&lt;/div&gt;

css:

div {float:left; width:50%}
div label, div input {display:block}

Without the extra div, it will be quite difficult to get the floated columns especially older IEs are concerned.
